packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
> Automatic transforms like the internal block-scoping and function hoisting have been a huge pain for people for this same reason

Those are actually changing the logic of the code. This is normalizing an AST into a coherent interface. We're [about to allow](https://github.com/babel/babel/pull/5856#discussion_r127376764) nameless declarations into `babel-types`, and this bug will only get worse. If we don't fix it here, there's no where else to do it and every transform now has to worry about a stupid edge case.